lib/pump.c

```diff
@@ -0,0 +1,167 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+//Number of sensors
+#define NUM_SENSORS 5
+
+//ID of the water level sensor
+#define ID_WATER_LEVEL_SENSOR 6
+
+//Lowest Humidicity level accepted for the water level sensor
+#define HUMIDICITY_LEVEL_ACCEPTED 5
+
+//Border values for the sensors we should edit the values
+#define PUMP_THRESHOLD_VERYHIGH  20
+#define PUMP_THRESHOLD_HIGH  15
+#define PUMP_THRESHOLD_LOW  10
+#define PUMP_THRESHOLD_VERYLOW  5
+
+//Struct that stores id of the sensors and humidity value
+ struct DataStruct
+  {
+      int id;
+      int data;
+  } pump_data;
+
+//Table to storage sensors data the size of the table depens on the number of sensors
+int table [5][2];
+bool pumpe_is_on = false;
+
+//Each time we open/close the pump we should reset the table values
+void reset_table( int table[][2])
+{
+
+	for (int i=0;i<NUM_SENSORS;i++){
+		table[i][0]=0;
+		table[i][1]=0;
+	}
+}
+
+//Function that prints the table with the values just for visual information
+void print_table( int table[][2])
+{
+
+	for(int i=0;i<NUM_SENSORS;i++){
+		printf("id: %d value: %d \n",table[i][0],table[i][1]);
+	}
+}
+
+
+bool pump_set_data(struct DataStruct pump_data, bool pumpe_is_on)
+{
+
+    int open_pumpe = 0;
+    int close_pumpe = 0;
+    int sum_hum = 0;
+    int avg_hum = 0;
+
+    if(pump_data.id == ID_WATER_LEVEL_SENSOR && pump_data.data < HUMIDICITY_LEVEL_ACCEPTED)
```

Comment:
This is the coindition i added to control if the sensor is the one that indicates the water level,we can replace the printf "need to be filled"  for a function that sends to the internet the alert that we're out of water